### PR TITLE
Display friendly credential names

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #2604.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7067/

File: jujugui/static/gui/src/app/store/env/controller-api.js [hunk 1]
```diff
@@ -1121,7 +1137,12 @@ YUI.add('juju-controller-api', function(Y) {
           const names = credentialTags.map(credentialTag => {
             return tags.parse(tags.CREDENTIAL, credentialTag);
           });
-          return {names: names};
+          const displayNames = names.map(
```
Comment (frankban) [inline jujugui/static/gui/src/app/store/env/controller-api.js]: Please update the docstring of this method to reflect the presence of a new field.

File: jujugui/static/gui/src/app/store/env/controller-api.js [hunk 2]
```diff
@@ -1192,6 +1213,7 @@ YUI.add('juju-controller-api', function(Y) {
           prev[name] = {
             authType: entry['auth-type'] || '',
             attrs: entry.attrs || {},
+            displayName: this._parseCredentialName(name),
```
Comment (frankban) [inline jujugui/static/gui/src/app/store/env/controller-api.js]: Ditto: update docstring by including displayName.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7068/

**Comment by kadams54** [comment]
QA is OK, but we now have another place we display credentials: in the expanded row of the user profile's model list. It would be nice to have a follow-up card that brings the displayName to the user profile as well. I'm not sure the best way to do that: call `_parseCredentialName` when storing the name with the model in `listModelInfo`? That has the disadvantage of the model not having the full credential name/id. Pass `_parseCredentialName` up to the `UserProfileEntity` component? Then the ID can be parsed into a user-friendly version at render time, but the downside is having to pass another prop into a deeply-buried component.

**Comment by huwshimi** [comment]
@kadams54 good spot, I've added an issue to keep track of it here: #2711.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7076/

**Comment by huwshimi** [comment]
Thanks @kadams54 @frankban :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**